docs/docs/secrets-managers.mdx

```diff
@@ -0,0 +1,142 @@
+---
+id: secrets-managers
+sidebar_label: Secrets Managers
+title: Secrets Managers
+description: Credentials used to authenticate to external services are safely stored in a secrets manager.
+abstract: Credentials managers are used to safeguard credentials used to authenticate to external services.
+---
+
+<!-- this file is version specific, do not use `@site/...` syntax -->
+import RasaProLabel from '@theme/RasaProLabel';
+import RasaProBanner from '@theme/RasaProBanner';
+
+<RasaProLabel />
+<RasaProBanner/>
+
+## Secrets Managers
+Secrets managers are used to safeguard credentials used to authenticate to external services,
+like databases.
+
+Rasa Pro supports the following secrets managers:
+* [HashiCorp Vault](https://www.hashicorp.com/products/vault)
+
+
+Currently, Rasa Pro supports sefeguarding credentials for the following services:
+- [Tracker Stores](./tracker-stores.mdx)
+
+
+## Vault Secrets Manager
+
+The Vault Secrets Manager is used to store credentials used to authenticate to external services.
+The credentials are stored in a Vault instance and can be encrypted at rest.
+
+If you need to encrypt your credentials,
+you can use the [Vault Transit Engine](https://www.vaultproject.io/docs/secrets/transit).
+
+:::note
+Expiring tokens need to be renewed periodically, and renewal process is done over the network,
+we will try to renew the token 5 seconds before it expires.
+If token's TTL is less than 5 seconds, we will try to renew it ASAP,
```

Comment:
```suggestion
If the token's time-to-live (TTL) is less than 5 seconds, we will try to renew it as soon as possible,
```

Reply:
what does "as soon as possible" mean in this context?

Reply:
Now I have realized that we have not covered this use case in the best way. 
If token's TTL is less than 5 seconds we should renew the token right away instead of waiting for token's remaining TTL.

Reply:
I opened [PR](https://github.com/RasaHQ/rasa-plus/pull/365) to address this situation.

Reply:
After aforementioned PR is merged, we can say that it will be renewed after 1 second.